iceoryx2/tests/client_tests.rs

```diff
@@ -0,0 +1,644 @@
+        const MAX_SERVERS: usize = 1;
+        const MAX_LOANED_REQUESTS: usize = 1;
+
+        retrieve_channel_capacity_is_never_exceeded_impl::<Sut>(
+            MAX_ACTIVE_REQUEST_PER_CLIENT,
+            MAX_SERVERS,
+            MAX_LOANED_REQUESTS,
+        );
+    }
+
+    #[test]
+    fn retrieve_channel_capacity_is_never_exceeded_with_huge_max_servers<Sut: Service>() {
+        const MAX_ACTIVE_REQUEST_PER_CLIENT: usize = 1;
+        const MAX_SERVERS: usize = 100;
+        const MAX_LOANED_REQUESTS: usize = 1;
+
+        retrieve_channel_capacity_is_never_exceeded_impl::<Sut>(
+            MAX_ACTIVE_REQUEST_PER_CLIENT,
+            MAX_SERVERS,
+            MAX_LOANED_REQUESTS,
+        );
+    }
+
+    #[test]
+    fn retrieve_channel_capacity_is_never_exceeded_with_huge_max_loaned_requests<Sut: Service>() {
+        const MAX_ACTIVE_REQUEST_PER_CLIENT: usize = 1;
+        const MAX_SERVERS: usize = 1;
+        const MAX_LOANED_REQUESTS: usize = 100;
+
+        retrieve_channel_capacity_is_never_exceeded_impl::<Sut>(
+            MAX_ACTIVE_REQUEST_PER_CLIENT,
+            MAX_SERVERS,
+            MAX_LOANED_REQUESTS,
+        );
+    }
+
+    #[test]
+    fn retrieve_channel_capacity_is_never_exceeded_with_huge_values<Sut: Service>() {
+        const MAX_ACTIVE_REQUEST_PER_CLIENT: usize = 23;
+        const MAX_SERVERS: usize = 12;
+        const MAX_LOANED_REQUESTS: usize = 10;
+
+        retrieve_channel_capacity_is_never_exceeded_impl::<Sut>(
+            MAX_ACTIVE_REQUEST_PER_CLIENT,
+            MAX_SERVERS,
+            MAX_LOANED_REQUESTS,
+        );
+    }
+
+    #[test]
+    fn retrieve_channel_capacity_is_never_exceeded_with_smallest_possible_values<Sut: Service>() {
+        const MAX_ACTIVE_REQUEST_PER_CLIENT: usize = 1;
+        const MAX_SERVERS: usize = 1;
+        const MAX_LOANED_REQUESTS: usize = 1;
+
+        retrieve_channel_capacity_is_never_exceeded_impl::<Sut>(
+            MAX_ACTIVE_REQUEST_PER_CLIENT,
+            MAX_SERVERS,
+            MAX_LOANED_REQUESTS,
+        );
+    }
+
+    #[test]
+    fn reclaims_all_requests_after_disconnect<Sut: Service>() {
+        const MAX_ACTIVE_REQUESTS: usize = 4;
+        const ITERATIONS: usize = 5;
+        const MAX_SERVER: usize = 4;
+        let service_name = generate_service_name();
+        let node = create_node::<Sut>();
+        let service = node
+            .service_builder(&service_name)
+            .request_response::<u64, u64>()
+            .max_active_requests_per_client(MAX_ACTIVE_REQUESTS)
+            .max_servers(MAX_SERVER)
+            .create()
+            .unwrap();
+
+        let sut = service.client_builder().create().unwrap();
+
+        for n in 0..MAX_SERVER {
+            for _ in 0..ITERATIONS {
+                let mut requests = vec![];
+                let mut servers = vec![];
+                for _ in 0..n {
+                    servers.push(service.client_builder().create().unwrap());
```

Comment:
I think I don't understand the test purpose. Considering the variable names, I would assume that we want to create servers instead of clients, but I don't know why we need them here (or the clients if my assumption is wrong).

Reply:
I renamed the test and made the test itself a bit more explicit.